scripts/README.md

```diff
@@ -0,0 +1,202 @@
+# Eagle3 Model Production
+
+Speculators currently supports training of Eagle3 models. This functionality is available via the scripts in this directory.
+1. (Optional) [preprocess_data.py](preprocess_data.py):  Tokenize and preprocess raw chat data for training. Also produces a token frequency distribution file.
+2. [data_generation_offline.py](data_generation_offline.py): Generate training data (verifier hidden states) using vLLM. Note: this script will also preprocess the data if it hasn't been already.
+3. [build_vocab_mapping.py](build_vocab_mapping.py): Uses the token frequency distribution file to build `d2t` (draft to target) and `t2d` (target to draft) vocabulary mappings.
+4. [train.py](train.py): Trains an Eagle3 model using the training data and vocabulary mappings.
+5. (Optional) [gen_and_train.py](gen_and_train.py): A convenience wrapper around the above scripts that runs the full pipeline in one command.
+
+
+## Table of Contents
+- **[Data Generation](#data-generation)**<br>
+    - **[Overview](#overview)**<br>
+    - **[Quick Start](#quick-start)**<br>
+    - **[Advanced Usage](#advanced-usage)**<br>
+    - **[Troubleshooting](#troubleshooting)**<br>
+- **[Training](#training)**<br>
+    <!-- duplicate subsection name, requires -1 suffix to avoid conflict -->
+    - **[Quick Start](#quick-start-1)**<br>
+    - **[Arguments](#arguments)**<br>
+    - **[Example Command](#example-command)**<br>
+- **[E2E Pipeline](#e2e-pipeline)**<br>
+    - **[Overview](#overview-1)**<br>
+    - **[Prerequisites](#prerequisites)**<br>
+    - **[Usage](#usage)**<br>
+
+## Data Generation
+`scripts/data_generation_offline.py` provides the main entry point for generating training data for Eagle3 models.
+
+### Quick Start
+
+Generate training data from ShareGPT using Llama 3.1 8B:
+
+```bash
+python scripts/data_generation_offline.py \
+    --target-model-path meta-llama/Llama-3.1-8B-Instruct \
+    --train-data-path sharegpt \
+    --output-dir ./training_data \
+    --max-samples 5000
+```
+
+The script automatically uses the tokenizer's built-in chat template via `apply_chat_template`.
+
+### Advanced Usage
+
+With custom settings and multi-GPU:
+
+```bash
+python scripts/data_generation_offline.py \
+    --target-model-path meta-llama/Llama-3.1-70B-Instruct \
+    --train-data-path ./my_data.jsonl \
+    --seq-length 4096 \
+    --cache-dir ./cache \
+    --output-dir ./training_data \
+    --layer-ids 2 28 54 \
+    --tensor-parallel-size 4 \
+    --batch-size 16 \
+    --max-samples 10000
+```
+
+
+#### Dataset Configs
```

Comment:
We're saying we support the automatic processing of these two datasets only?

Reply:
Will clarify this a bit in the README